### PR TITLE
Minor: Move `function::Hint` to `datafusion-expr` crate to avoid physical-expr dependency for `datafusion-function` crate

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Closes #.

## Rationale for this change

As titile

### CODE REVIEW

**Review by goldmedal — APPROVED** [review]
Thanks @jayzhan211. LGTM

File: datafusion/functions/Cargo.toml [hunk 1]
```diff
@@ -73,7 +73,6 @@ chrono = { workspace = true }
 datafusion-common = { workspace = true }
 datafusion-execution = { workspace = true }
 datafusion-expr = { workspace = true }
-datafusion-physical-expr = { workspace = true, default-features = true }
```
Comment (alamb) [inline datafusion/functions/Cargo.toml:76]: 🎉

File: datafusion/expr/src/function.rs [hunk 2]
```diff
@@ -23,6 +23,14 @@ use arrow::datatypes::{DataType, Field, Schema};
 use datafusion_common::Result;
 use std::sync::Arc;
 
+#[derive(Debug, Clone, Copy)]
```
Comment (alamb) [inline datafusion/expr/src/function.rs:26]: Eventually it would be great to add some more comments here, but maybe as a follow on PR

**Review by alamb — APPROVED** [review]
Thank you @jayzhan211

**Comment by jayzhan211** [comment]
Thanks @alamb and @goldmedal

**Pull request merged**